sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -687,7 +688,13 @@ case class ReplaceTableAsSelect(
     isAnalyzed: Boolean = false)
   extends V2CreateTableAsSelectPlan {
 
-  override def markAsAnalyzed(ac: AnalysisContext): LogicalPlan = copy(isAnalyzed = true)
+  override def markAsAnalyzed(ac: AnalysisContext): LogicalPlan = {
+    // RTAS may drop and recreate table before query execution, breaking self-references
```

Comment:
This seems to be one of a kind. You have to first drop the table and then create a new nested QE to do insert.